redisearch/client_test.go

```diff
@@ -553,33 +578,54 @@ func TestClient_AddField(t *testing.T) {
 	assert.Nil(t, err)
 	err = c.Index(NewDocument("doc-n1", 1.0).Set("age", 15))
 	assert.Nil(t, err)
+	teardown(c)
 }
 
-func TestClient_CreateIndex(t *testing.T) {
-	type fields struct {
-		pool ConnPool
-		name string
-	}
-	type args struct {
-		s *Schema
-	}
-	tests := []struct {
-		name    string
-		fields  fields
-		args    args
-		wantErr bool
-	}{
-		// TODO: Add test cases.
-	}
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			i := &Client{
-				pool: tt.fields.pool,
-				name: tt.fields.name,
-			}
-			if err := i.CreateIndex(tt.args.s); (err != nil) != tt.wantErr {
-				t.Errorf("CreateIndex() error = %v, wantErr %v", err, tt.wantErr)
-			}
-		})
+func TestClient_GetRediSearchVersion(t *testing.T) {
+	c := createClient("version-test")
+	_, err := c.GetRediSearchVersion()
+	assert.Nil(t, err)
+}
+
+func TestClient_CreateIndexWithIndexDefinition(t *testing.T) {
+	i := createClient("index-definition-test")
+	version, err := i.GetRediSearchVersion()
+	assert.Nil(t, err)
+	if version >= 20000 {
+
+		type args struct {
+			schema     *Schema
+			definition *IndexDefinition
+		}
+		tests := []struct {
+			name    string
+			args    args
+			wantErr bool
+		}{
+			{"no-indexDefinition", args{NewSchema(DefaultOptions).
+				AddField(NewTextField("name")).
+				AddField(NewTextField("addr")), nil}, false},
+			{"default-indexDefinition", args{NewSchema(DefaultOptions).
+				AddField(NewTextField("name")).
+				AddField(NewTextField("addr")), NewIndexDefinition()}, false},
+			{"score-indexDefinition", args{NewSchema(DefaultOptions).
+				AddField(NewTextField("name")).
+				AddField(NewTextField("addr")), NewIndexDefinition().SetScore(0.25)}, false},
+			{"language-indexDefinition", args{NewSchema(DefaultOptions).
+				AddField(NewTextField("name")).
+				AddField(NewTextField("addr")), NewIndexDefinition().SetLanguage("portuguese")}, false},
+			{"language_field-indexDefinition", args{NewSchema(DefaultOptions).
+				AddField(NewTextField("name")).
+				AddField(NewTextField("lang")).
+				AddField(NewTextField("addr")), NewIndexDefinition().SetLanguageField("lang")}, false},
```

Comment:
Maybe we should also add test for:
SetScoreField, SetPayloadField, and set the prefix to follow and maybe also filter?
What do you think?

Reply:
agree. doing so :+1: